### ISSUE TITLE
Navigation issue

### ISSUE THREAD

**Issue by sequelpro** [sequelpro/sequelpro#1149]
Issue imported from Google Code: https://code.google.com/p/sequel-pro/issues/detail?id=1149
Reporter:possibil...@gmail.com Date:2011-08-15 04:10:10 Status:Accepted

<b>What steps will reproduce the problem?</b>
1. Look at Content table with foreign key references that reference to a primary key
2. Click Navigation error next to field which indicates (null)
3. Observe empty content table with filter set to IS null and no easy way to remove filter

<b>What is the expected output? What do you see instead?</b>

When looking at the content table I can navigate using the arrows from rows in my current table to other tables via foreign key constraints. This is a nice navigation feature.

However this feature also ALLOWS inappropriate attempts to navigate. The main one is if the row contains a null in the foreign key constrained column and I am navigating via a foreign key constraint to another table via a primary key with NOT NULL set (most frequently this IS the case) then the arrow is not going to get me anything in the next table because this is by definition impossible without need to even query the table. SO the arrow in the column should NOT be displayed for null content in that column, only for non-null values.

And this is where the first bug comes in. If I am stupid enough to click on the arrow next to the null column it attempts to go to the next table with a IS NULL filter. Which gets me no content (duh). The bug is now not only do I have no content but there is no easy way to remove the filter like there is with a value in the filter bar. The X on the right end of the filter bar is missing when the criteria is &quot;IS NULL&quot; so I have to navigate somewhere or change the criteria or do something to get rid of the filter for this table.

And here is the subsequent bug - if I have the criteria as IS null and move to another table and then back to this table it remembers the filter. Nice feature, but recall this is a filter I cannot get rid of easily. IN fact if I change the criteria to something else and enter in a value the X pops up to allow me to not filter in the filter bar. BUT if you change the filter criteria to IS NULL now the value I entered is greyed out AND the X while showing is DISABLED!

So basically if you change the filter to is null good luck simply using the X on the filter bar to stop filtering.

<b>What version of Sequel Pro are you using? What version of MySQL are you</b>
connecting to on the server? Nightly build 3386, Mysql 5.0.92

<b>Please provide any additional information below.</b>
I think this is not intuitive and in the spirit of letting the user not shoot themselves ion the foot, and if they do at least make it easy for them to undo their action this should be reconsidered.
